File: Java/Technology Fundamentals with Java/Exercises Java Code/src/ArraysExercise/ZigZagArrays.java
package ArraysExercise;

import java.util.Arrays;
import java.util.Scanner;

public class ZigZagArrays {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());

        int[] firstArray = new int[n];
        int[] secondArray = new int[n];

        for (int i = 0; i < n ; i++) {
            int[] intArray = Arrays.stream(sc.nextLine().split(" "))
                    .mapToInt(Integer::parseInt).toArray();
            if (i%2==0){
                firstArray[i] = intArray[0];
                secondArray[i] = intArray[1];
            } else {
                firstArray[i] = intArray[1];
                secondArray[i] = intArray[0];
            }
        }
        for (int number: firstArray) {
            System.out.print(number + " ");
        }
        System.out.println();
        for (int number: secondArray) {
            System.out.print(number + " ");
        }
        System.out.println();
    }
}

File: C#/Technology Fundamentals with C#/02. Arrays - Exercise/03. Zig-Zag Arrays/ZigZagArrays.cs
using System;
using System.Linq;

public class ZigZagArrays
{
    public static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        int[] firstArray = new int[n];
        int[] secondArray = new int[n];

        for (int i = 0; i < n; i++)
        {
            int[] input = Console.ReadLine().Split().Select(int.Parse).ToArray();

            if (i % 2 == 0)
            {
                firstArray[i] = input[0];
                secondArray[i] = input[1];
            }
            else
            {
                firstArray[i] = input[1];
                secondArray[i] = input[0];
            }
        }

        foreach (var number in firstArray)
        {
            Console.Write(number + " ");
        }

        Console.WriteLine();

        foreach (var number in secondArray)
        {
            Console.Write(number + " ");
        }

        Console.WriteLine();
    }
}